qutip/tests/core/test_qobjevo.py

```diff
@@ -42,111 +42,442 @@
+    cplx_coeff = Base_coeff(_cplx, "exp(1j*t*w2)")
+    qobjevos = {"qobj": rand_herm(N), "scalar":np.random.rand()}
+    cplx_forms = ['func', 'string', 'spline', 'array', 'array_log']
+    mixed_forms = ['with_args', 'no_args', 'mixed_tlist']
+
+    def _rand_t(self):
+        return np.random.rand() * 0.9 + 0.05
+
+    def test_creation(self, form):
+        if form == "cte":
+            obj = QobjEvo(self.cte_qobj)
+        elif form == "func":
+            obj = QobjEvo([self.cte_qobj,
+                           [self.cplx_qobj, self.cplx_coeff.func]],
+                          args=self.args)
+        elif form == "string":
+            obj = QobjEvo([self.cte_qobj,
+                           [self.cplx_qobj, self.cplx_coeff.string]],
+                          args=self.args)
+        elif form == "spline":
+            obj = QobjEvo([[self.cplx_qobj,
+                            self.cplx_coeff.spline(self.tlist, self.args)],
+                           self.cte_qobj])
+        elif form == "array":
+            obj = QobjEvo([self.cte_qobj,
+                           [self.cplx_qobj,
+                            self.cplx_coeff.array(self.tlist, self.args)]],
+                          tlist=self.tlist)
+        elif form == "array_log":
+            obj = QobjEvo([self.cte_qobj,
+                           [self.cplx_qobj,
+                            self.cplx_coeff.array(self.tlistlog, self.args)]],
+                          tlist=self.tlistlog)
+        elif form == "with_args":
+            obj = QobjEvo([self.cte_qobj,
+                           [self.cplx_qobj, self.cplx_coeff.func],
+                           [self.real_qobj, self.real_coeff.string]],
+                          args=self.args)
+        elif form == "no_args":
+            obj = QobjEvo([self.cte_qobj,
+                           [self.cplx_qobj, self.cplx_coeff.spline(self.tlist,
+                                                                   self.args)],
+                           [self.real_qobj, self.real_coeff.array(self.tlist,
+                                                                  self.args)]],
+                          tlist=self.tlist)
+        elif form == "mixed_tlist":
+            coeff_1 = coefficient(self.cplx_coeff.array(self.tlist,
+                                                        self.args),
+                                  tlist=self.tlist)
+            coeff_2 = coefficient(self.real_coeff.array(self.tlistlog,
+                                                        self.args),
+                                  tlist=self.tlistlog)
+            obj = QobjEvo([self.cte_qobj,
+                           [self.cplx_qobj, coeff_1],
+                           [self.real_qobj, coeff_2]])
+        self.qobjevos[form] = obj
+
+    def test_call(self, form):
+        t = self._rand_t()
+        assert isinstance(self.qobjevos[form](t), Qobj)
+
+        for _ in range(10):
```

Comment:
You can use `pytest.mark.repeat(10)` as a decorator on the test, then we get stats on if all runs failed, or just one.

Reply:
I don't like `mark.repeat`, it would create 10 lines of tests for each type of `QobjEvo`. It clogs the pytest outputs more than give extra useful information...

Reply:
I wrote the `mark.repeat` code back when I was first starting to learn `pytest` - it's in our `conftest.py`.  We could probably change it to only list failing lines when reporting the tests.  I'd argue though that it does add information, because you find out whether it's a systematic problem with the test (all repeats fail) or if it's a one-off, and `pytest` collates the failures down at the end, where extra repeats don't show at all, so there's no textual cost in the main report.